### PR TITLE
python3Packages.synergy: init at 0.5.1

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Really interesting python module i wanted to check out.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by kira64xyz — APPROVED** [review]

File: pkgs/development/python-modules/synergy/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  checkInputs = [ pytestCheckHook ];
+  pythonImportsCheck = [ "synergy" ];
+
+  meta = with lib; {
+    description = "A Python library for calculating, analyzing, and visualizing drug combination synergy";
+    # Upstream readme asks for this to be cited.
+    longDescription = ''
+      Wooten, David J, and Albert, Réka. synergy - A Python library for calculating, analyzing, and visualizing drug combination synergy.
+      (2020) Bioinformatics. https://doi.org/10.1093/bioinformatics/btaa826
+    '';
```
Comment (fabaff) [inline pkgs/development/python-modules/synergy/default.nix:43]: Citations are usually for publications. The users are using the module. Nixpkgs is simply providing it. Thus `longDescription` should not be use for the citation. This should a task the users have to take care.

Comment (IvarWithoutBones) [reply pkgs/development/python-modules/synergy/default.nix]: Fair enough, removed it :+1:

**Comment by IvarWithoutBones** [comment]
Result of `nixpkgs-review pr 131131` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.synergy</li>
    <li>python39Packages.synergy</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131131` at 4ebfdf3b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/cb44a41fb483f9b7d7a52e47ea30150e">python38Packages.synergy</a></li>
    <li><a href="https://gist.github.com/d3c2a184602f240630802e5c919957ff">python39Packages.synergy</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by risicle** [comment]
Nitpick, but I don't *think* this is gpl3 *plus* - I can't see any evidence of the "or greater" clause in https://github.com/djwooten/synergy/blob/master/LICENSE

**Comment by risicle** [comment]
Builds happily macos 10.15.

**Comment by IvarWithoutBones** [comment]
At the bottom at the projects readme, they state that the project is licensed under the GPL3 or any later revision.

**Pull request merged**